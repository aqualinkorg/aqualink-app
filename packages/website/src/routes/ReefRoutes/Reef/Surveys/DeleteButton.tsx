import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  withStyles,
  WithStyles,
  createStyles,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  Typography,
  IconButton,
  Button,
  Collapse,
  LinearProgress,
} from "@material-ui/core";
import Alert from "@material-ui/lab/Alert";
import DeleteOutlineIcon from "@material-ui/icons/DeleteOutline";
import CloseIcon from "@material-ui/icons/Close";
import moment from "moment";

import surveyServices from "../../../../services/surveyServices";
import { userInfoSelector } from "../../../../store/User/userSlice";
import { surveysRequest } from "../../../../store/Survey/surveyListSlice";

const DeleteButton = ({
  reefId,
  surveyId,
  diveDate,
  classes,
}: DeleteButtonProps) => {
  const user = useSelector(userInfoSelector);
  const dispatch = useDispatch();
  const [open, setOpen] = useState<boolean>(false);
  const [alertOpen, setAlertOpen] = useState<boolean>(false);
  const [alertText, setAlertText] = useState<string>("");
  const [loading, setLoading] = useState<boolean>(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
    setAlertOpen(false);
    setAlertText("");
  };

  const onSurveyDelete = () => {
    if (surveyId && user && user.token) {
      setLoading(true);
      surveyServices
        .deleteSurvey(surveyId, user.token)
        .then(() => {
          dispatch(surveysRequest(`${reefId}`));
        })
        .catch((error) => {
          setAlertOpen(true);
          setAlertText(error.message);
        })
        .finally(() => setLoading(false));
    }
  };

  return (
    <>
      <IconButton onClick={handleClickOpen}>
        <DeleteOutlineIcon color="primary" />
      </IconButton>
      <Dialog open={open} onClose={handleClose}>
        <DialogTitle className={classes.dialogTitle}>
          <Typography color="textSecondary">
            Are you sure you would like to delete the survey for{" "}
            {moment(diveDate).format("MM/DD/YYYY")}?
          </Typography>
        </DialogTitle>
        <DialogContent className={classes.dialogContent}>
          {loading && <LinearProgress />}
          <Collapse className={classes.alert} in={alertOpen}>
            <Alert
              severity="error"
              action={
                <IconButton
                  color="inherit"
                  size="small"
                  onClick={() => {
                    setAlertOpen(false);
                  }}
                >
                  <CloseIcon fontSize="inherit" />
                </IconButton>
              }
            >
              {alertText}
            </Alert>
          </Collapse>
        </DialogContent>
        <DialogActions>
          <Button
            size="small"
            variant="contained"
            color="secondary"
            onClick={handleClose}
          >
            No
          </Button>
          <Button
            size="small"
            variant="contained"
            color="primary"
            onClick={onSurveyDelete}
          >
            Yes
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

const styles = () =>
  createStyles({
    dialogTitle: {
      backgroundColor: "#F4F4F4",
    },
    dialogContent: {
      padding: 0,
    },
    alert: {
      width: "100%",
    },
  });

interface DeleteButtonIncomingProps {
  surveyId?: number;
  diveDate?: string | null;
  reefId: number;
}

type DeleteButtonProps = DeleteButtonIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(DeleteButton);
