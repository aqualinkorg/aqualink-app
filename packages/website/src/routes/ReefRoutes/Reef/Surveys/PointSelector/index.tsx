import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Grid,
  Typography,
  MenuItem,
  Select,
  IconButton,
  OutlinedInput,
} from "@material-ui/core";
import { Create, DeleteOutline } from "@material-ui/icons";

import { Pois } from "../../../../../store/Reefs/types";
import { EditPoiNameDraft } from "../types";
import EditDialog, { Action } from "../../../../../common/Dialog";

const PointSelector = ({
  mountPois,
  pointOptions,
  point,
  editPoiNameDraft,
  isReefAdmin,
  editPoiNameLoading,
  onChangePoiName,
  handlePointChange,
  toggleEditPoiNameEnabled,
  submitPoiNameUpdate,
  onDeleteButtonClick,
  classes,
}: PointSelectorProps) => {
  const [editDialogOpen, setEditDialogOpen] = useState(false);
  const [editPoi, setEditPoi] = useState<Pois>();

  const onEditDialogClose = useCallback(() => {
    toggleEditPoiNameEnabled(false, editPoi?.id);
    setEditPoi(undefined);
    setEditDialogOpen(false);
  }, [editPoi, toggleEditPoiNameEnabled]);

  const onEditPoiSubmit = useCallback(() => {
    if (editPoi) {
      submitPoiNameUpdate(editPoi.id);
    }
  }, [editPoi, submitPoiNameUpdate]);

  useEffect(() => {
    if (!editPoiNameLoading) {
      setEditPoi(undefined);
      setEditDialogOpen(false);
    }
  }, [editPoiNameLoading]);

  const editDialogActions: Action[] = [
    {
      size: "small",
      variant: "contained",
      color: "secondary",
      text: "Close",
      action: onEditDialogClose,
    },
    {
      size: "small",
      variant: "contained",
      color: "primary",
      text: editPoiNameLoading ? "Updating..." : "Save",
      action: onEditPoiSubmit,
      disabled: editPoiNameLoading,
    },
  ];

  return (
    <>
      {editPoi && (
        <EditDialog
          actions={editDialogActions}
          open={editDialogOpen}
          header={editPoi.name || ""}
          onClose={onEditDialogClose}
          content={
            <OutlinedInput
              autoFocus
              className={classes.editPoiTextField}
              fullWidth
              value={editPoiNameDraft[editPoi.id]}
              onChange={onChangePoiName(editPoi.id)}
              error={editPoiNameDraft[editPoi.id] === ""}
            />
          }
        />
      )}
      <Grid container alignItems="center" item md={12} lg={4}>
        <Grid item>
          <Typography variant="h6" className={classes.subTitle}>
            Survey Point:
          </Typography>
        </Grid>
        {mountPois && (
          <Grid item>
            <Select
              className={classes.selector}
              labelId="survey-point"
              id="survey-point"
              name="survey-point"
              value={
                pointOptions.map((item) => item.name).includes(point)
                  ? point
                  : "All"
              }
              onChange={handlePointChange}
              onClose={() => toggleEditPoiNameEnabled(false)}
              renderValue={(selected) => selected as string}
            >
              <MenuItem value="All">
                <Typography className={classes.menuItem} variant="h6">
                  All
                </Typography>
              </MenuItem>
              {pointOptions.map(
                (item) =>
                  item.name !== null && (
                    <MenuItem
                      className={classes.menuItem}
                      value={item.name}
                      key={item.id}
                    >
                      <Grid
                        container
                        alignItems="center"
                        justify="space-between"
                        spacing={1}
                      >
                        <Grid className={classes.itemName} item>
                          {item.name}
                        </Grid>
                        {isReefAdmin && (
                          <Grid item>
                            <Grid container item spacing={1}>
                              <Grid item>
                                <IconButton
                                  className={classes.menuButton}
                                  onClick={(event) => {
                                    setEditDialogOpen(true);
                                    setEditPoi(item);
                                    event.stopPropagation();
                                  }}
                                >
                                  <Create color="primary" />
                                </IconButton>
                              </Grid>
                              <Grid item>
                                <IconButton
                                  className={classes.menuButton}
                                  onClick={(event) => {
                                    onDeleteButtonClick(item.id);
                                    event.stopPropagation();
                                  }}
                                >
                                  <DeleteOutline color="primary" />
                                </IconButton>
                              </Grid>
                            </Grid>
                          </Grid>
                        )}
                      </Grid>
                    </MenuItem>
                  )
              )}
            </Select>
          </Grid>
        )}
      </Grid>
    </>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    subTitle: {
      lineHeight: 1,
      color: "#474747",
      marginRight: "1rem",
    },
    selector: {
      minWidth: 120,
      maxWidth: 240,
      color: theme.palette.primary.main,
    },
    itemName: {
      maxWidth: 140,
      whiteSpace: "normal",
      overflowWrap: "break-word",
      wordWrap: "break-word",
      wordBreak: "break-word",
      hyphens: "auto",
    },
    menuItem: {
      minWidth: 200,
      color: theme.palette.primary.main,
    },
    editPoiTextField: {
      color: "black",
      height: "2.5rem",
      alignItems: "center",
    },
    menuButton: {
      padding: 0,
    },
    checkIcon: {
      color: theme.palette.success.main,
    },
    closeIcon: {
      color: theme.palette.error.main,
    },
  });

interface PointSelectorIncomingProps {
  mountPois: boolean;
  pointOptions: Pois[];
  point: string;
  editPoiNameDraft: EditPoiNameDraft;
  isReefAdmin: boolean;
  editPoiNameLoading: boolean;
  onChangePoiName: (
    key: number
  ) => (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
  handlePointChange: (event: ChangeEvent<{ value: unknown }>) => void;
  toggleEditPoiNameEnabled: (
    enabled: boolean,
    key?: number | undefined
  ) => void;
  submitPoiNameUpdate: (key: number) => void;
  onDeleteButtonClick: (id: number) => void;
}

type PointSelectorProps = PointSelectorIncomingProps &
  WithStyles<typeof styles>;

export default withStyles(styles)(PointSelector);
