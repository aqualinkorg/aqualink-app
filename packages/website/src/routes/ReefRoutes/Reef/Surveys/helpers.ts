import { SurveyListState, Observations } from "../../../../store/Survey/types";
import { sortByDate } from "../../../../helpers/sortDailyData";

const filterSurveys = (
  list: SurveyListState["list"],
  observation: Observations | "any",
  point: number
): SurveyListState["list"] => {
  const sortedSurveys = sortByDate(list, "diveDate", "desc");

  if (observation === "any" && point === -1) {
    return sortedSurveys;
  }
  if (observation === "any") {
    return sortedSurveys.filter((survey) =>
      survey.surveyPoints?.includes(point)
    );
  }
  if (point === -1) {
    return sortedSurveys.filter((survey) =>
      survey.observations?.includes(observation)
    );
  }
  return sortedSurveys.filter(
    (survey) =>
      survey.observations?.includes(observation) &&
      survey.surveyPoints?.includes(point)
  );
};

export default filterSurveys;
