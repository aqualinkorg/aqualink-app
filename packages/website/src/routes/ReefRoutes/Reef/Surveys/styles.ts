import theme from "../../../../layout/App/theme";

const styles = {
  dateWrapper: {
    marginBottom: "1rem",
  },
  dates: {
    fontWeight: 500,
    lineHeight: 0.81,
    color: "#757575",
  },
  addNewButton: {
    color: "#979797",
    height: "2rem",
    width: "2rem",
  },
  surveyCardWrapper: {
    marginBottom: "2rem",
  },
  surveyCard: {
    width: "80%",
    backgroundColor: theme.palette.primary.light,
    border: 1,
    borderStyle: "solid",
    borderColor: "#dddddd",
    borderRadius: 2,
    height: "16rem",
    [theme.breakpoints.down("sm")]: {
      height: "32rem",
    },
    [theme.breakpoints.down("xs")]: {
      width: "100%",
    },
  },
  cardImage: {
    height: "100%",
    width: "100%",
  },
  cardFields: {
    fontWeight: 500,
    lineHeight: 2,
    color: "#9ea6aa",
  },
  cardValues: {
    lineHeight: 2,
    color: "#2f2f2f",
  },
  valuesWithMargin: {
    marginLeft: "1rem",
  },
};

export default styles;
