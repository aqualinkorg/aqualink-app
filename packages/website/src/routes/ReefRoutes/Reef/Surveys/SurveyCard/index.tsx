import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Paper,
  Grid,
  CardMedia,
  Typography,
  Button,
} from "@material-ui/core";
import { Link } from "react-router-dom";

import { formatNumber } from "../../../../../helpers/numberUtils";
import DeleteButton from "../DeleteButton";
import { SurveyListItem } from "../../../../../store/Survey/types";
import incomingStyles from "../styles";

const SurveyCard = ({ isAdmin, reefId, survey, classes }: SurveyCardProps) => {
  return (
    <Paper elevation={0} className={classes.surveyCard}>
      <Grid style={{ height: "100%" }} container justify="space-between">
        <Grid className={classes.cardImageWrapper} item xs={12} md={5}>
          {survey.featuredSurveyMedia && (
            <CardMedia
              className={classes.cardImage}
              image={survey.featuredSurveyMedia.url}
            />
          )}
        </Grid>
        <Grid className={classes.infoWrapper} container item xs={12} md={7}>
          <Grid className={classes.info} container item xs={12}>
            {survey.userId!.fullName && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  User:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {survey.userId!.fullName}
                </Typography>
              </Grid>
            )}
            {survey.comments && (
              <Grid
                className={classes.commentsWrapper}
                container
                alignItems="flex-start"
                item
                xs={12}
              >
                <Grid style={{ height: "80%" }} item xs={12}>
                  <Typography className={classes.cardFields} variant="h6">
                    Comments:
                  </Typography>
                  <Typography
                    className={`${classes.cardValues} ${classes.comments}`}
                    variant="h6"
                  >
                    {survey.comments}
                  </Typography>
                </Grid>
              </Grid>
            )}
            {survey.temperature && (
              <Grid container alignItems="center" item xs={12}>
                <Typography className={classes.cardFields} variant="h6">
                  Temp:
                </Typography>
                <Typography
                  className={`${classes.cardValues} ${classes.valuesWithMargin}`}
                  variant="h6"
                >
                  {`${formatNumber(survey.temperature, 1)} Â°C`}
                </Typography>
              </Grid>
            )}
            <Grid
              container
              alignItems="center"
              justify="space-between"
              item
              xs={12}
            >
              <Grid item xs={10}>
                <Link
                  style={{ color: "inherit", textDecoration: "none" }}
                  to={`/reefs/${reefId}/survey_details/${survey.id}`}
                >
                  <Button size="small" variant="outlined" color="primary">
                    VIEW DETAILS
                  </Button>
                </Link>
              </Grid>
              {isAdmin && (
                <Grid container justify="flex-end" item xs={2}>
                  <DeleteButton
                    reefId={reefId}
                    surveyId={survey.id}
                    diveDate={survey.diveDate}
                  />
                </Grid>
              )}
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Paper>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    ...incomingStyles,
    cardImageWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    infoWrapper: {
      height: "100%",
      [theme.breakpoints.down("sm")]: {
        height: "50%",
      },
    },
    commentsWrapper: {
      maxHeight: "55%",
    },
    comments: {
      height: "100%",
      overflowY: "auto",
    },
    info: {
      height: "100%",
      padding: "0.5rem 0.5rem 0.5rem 1rem",
    },
  });

interface SurveyCardIncomingProps {
  isAdmin: boolean;
  reefId: number;
  survey: SurveyListItem;
}

type SurveyCardProps = SurveyCardIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(SurveyCard);
