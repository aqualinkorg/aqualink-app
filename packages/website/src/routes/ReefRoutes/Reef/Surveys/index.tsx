import React, { useState, useCallback, useEffect } from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Theme,
  Grid,
  Typography,
  Select,
  FormControl,
  MenuItem,
  Box,
  IconButton,
  Tooltip,
} from "@material-ui/core";
import DeleteOutlineIcon from "@material-ui/icons/DeleteOutline";
import { useDispatch, useSelector } from "react-redux";
import Axios from "axios";

import Timeline from "./Timeline";
import { userInfoSelector } from "../../../../store/User/userSlice";
import { surveysRequest } from "../../../../store/Survey/surveyListSlice";
import observationOptions from "../../../../constants/uploadDropdowns";
import { SurveyMedia } from "../../../../store/Survey/types";
import reefServices from "../../../../services/reefServices";
import { Pois } from "../../../../store/Reefs/types";
import { isAdmin } from "../../../../helpers/isAdmin";
import DeletePoiDialog, { Action } from "../../../../common/Dialog";
import { useBodyLength } from "../../../../helpers/useBodyLength";

const Surveys = ({ reefId, classes }: SurveysProps) => {
  const [point, setPoint] = useState<string>("All");
  const [pointOptions, setPointOptions] = useState<Pois[]>([]);
  const [deletePoiDialogOpen, setDeletePoiDialogOpen] = useState<boolean>(
    false
  );
  const [poiToDelete, setPoiToDelete] = useState<number | null>(null);
  const [mountPois, setMountPois] = useState<boolean>(false);
  const [observation, setObservation] = useState<
    SurveyMedia["observations"] | "any"
  >("any");
  const [windowWidth, setWindowWidth] = useState<number>(window.innerWidth);
  const user = useSelector(userInfoSelector);
  const isReefAdmin = isAdmin(user, reefId);
  const dispatch = useDispatch();

  const bodyLength = useBodyLength();

  useEffect(() => {
    const source = Axios.CancelToken.source();
    reefServices
      .getReefPois(`${reefId}`, source.token)
      .then((response) => {
        setPointOptions(response.data);
        setMountPois(true);
      })
      .catch((error) => {
        if (!Axios.isCancel(error)) {
          setMountPois(false);
        }
      });
    return () => {
      source.cancel();
    };
  }, [setPointOptions, reefId]);

  const onResize = useCallback(() => {
    setWindowWidth(window.innerWidth);
  }, []);

  useEffect(() => {
    window.addEventListener("resize", onResize);
    return () => {
      window.removeEventListener("resize", onResize);
    };
  }, [onResize]);

  const handlePointChange = (event: React.ChangeEvent<{ value: unknown }>) => {
    setPoint(event.target.value as string);
  };

  const handleObservationChange = (
    event: React.ChangeEvent<{ value: unknown }>
  ) => {
    setObservation(event.target.value as SurveyMedia["observations"] | "any");
  };

  const pointIdFinder = (name: string) => {
    return pointOptions.find((option) => option.name === name)?.id || -1;
  };

  const handleDeletePoiDialogClose = () => {
    setDeletePoiDialogOpen(false);
    setPoiToDelete(null);
  };

  const handleSurveyPointDelete = () => {
    if (user && user.token && poiToDelete) {
      reefServices
        .deleteReefPoi(poiToDelete, user.token)
        .then(() =>
          setPointOptions(
            pointOptions.filter((option) => option.id !== poiToDelete)
          )
        )
        .then(() => {
          dispatch(surveysRequest(`${reefId}`));
        })
        .then(() => {
          setDeletePoiDialogOpen(false);
          setPoiToDelete(null);
        });
    }
  };

  const deletePoiDialogActions: Action[] = [
    {
      size: "small",
      variant: "contained",
      color: "secondary",
      text: "No",
      action: handleDeletePoiDialogClose,
    },
    {
      size: "small",
      variant: "contained",
      color: "primary",
      text: "Yes",
      action: handleSurveyPointDelete,
    },
  ];

  return (
    <>
      <DeletePoiDialog
        open={deletePoiDialogOpen}
        onClose={handleDeletePoiDialogClose}
        header="Are you sure you want to delete this survey point? It will be deleted across all surveys."
        actions={deletePoiDialogActions}
      />
      <Grid className={classes.root} container justify="center" spacing={2}>
        <Box
          bgcolor="#f5f6f6"
          position="absolute"
          height="100%"
          width={bodyLength}
          zIndex="-1"
        />
        <Grid
          className={classes.surveyWrapper}
          container
          justify="space-between"
          item
          lg={12}
          xs={11}
          alignItems="baseline"
        >
          <Grid
            container
            justify={windowWidth < 1280 ? "flex-start" : "center"}
            item
            md={12}
            lg={4}
          >
            <Typography className={classes.title}>Survey History</Typography>
          </Grid>
          <Grid container alignItems="center" item md={12} lg={4}>
            <Grid item>
              <Typography variant="h6" className={classes.subTitle}>
                Survey Point:
              </Typography>
            </Grid>
            {mountPois && (
              <Grid item>
                <FormControl className={classes.formControl}>
                  <Select
                    labelId="survey-point"
                    id="survey-point"
                    name="survey-point"
                    value={point}
                    onChange={handlePointChange}
                    className={classes.selectedItem}
                    renderValue={(selected) => selected as string}
                  >
                    <MenuItem value="All">
                      <Typography className={classes.menuItem} variant="h6">
                        All
                      </Typography>
                    </MenuItem>
                    {pointOptions.map(
                      (item) =>
                        item.name !== null && (
                          <MenuItem
                            className={classes.menuItem}
                            value={item.name}
                            key={item.id}
                          >
                            <Grid
                              container
                              alignItems="center"
                              justify="space-between"
                            >
                              <Grid item>{item.name}</Grid>
                              {isReefAdmin && (
                                <Grid item>
                                  <Tooltip title="Delete this survey point">
                                    <IconButton
                                      className={classes.pointDeleteButton}
                                      onClick={(event) => {
                                        setDeletePoiDialogOpen(true);
                                        setPoiToDelete(item.id);
                                        event.stopPropagation();
                                      }}
                                    >
                                      <DeleteOutlineIcon color="primary" />
                                    </IconButton>
                                  </Tooltip>
                                </Grid>
                              )}
                            </Grid>
                          </MenuItem>
                        )
                    )}
                  </Select>
                </FormControl>
              </Grid>
            )}
          </Grid>
          <Grid
            container
            alignItems="center"
            justify={windowWidth < 1280 ? "flex-start" : "center"}
            item
            md={12}
            lg={4}
          >
            {/* TODO - Make observation a required field. */}
            <Grid item>
              <Typography variant="h6" className={classes.subTitle}>
                Observation:
              </Typography>
            </Grid>
            <Grid item>
              <FormControl className={classes.formControl}>
                <Select
                  labelId="survey-observation"
                  id="survey-observation"
                  name="survey-observation"
                  value={observation}
                  onChange={handleObservationChange}
                  className={classes.selectedItem}
                  inputProps={{ className: classes.textField }}
                >
                  <MenuItem value="any">
                    <Typography className={classes.menuItem} variant="h6">
                      Any
                    </Typography>
                  </MenuItem>
                  {observationOptions.map((item) => (
                    <MenuItem
                      className={classes.menuItem}
                      value={item.key}
                      key={item.key}
                    >
                      {item.value}
                    </MenuItem>
                  ))}
                </Select>
              </FormControl>
            </Grid>
          </Grid>
        </Grid>
        <Grid container justify="center" item xs={11} lg={12}>
          <Timeline
            isAdmin={isReefAdmin}
            reefId={reefId}
            observation={observation}
            point={pointIdFinder(point)}
          />
        </Grid>
      </Grid>
    </>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    root: {
      marginTop: "5rem",
      position: "relative",
    },
    surveyWrapper: {
      marginTop: "5rem",
    },
    title: {
      fontSize: 22,
      fontWeight: "normal",
      fontStretch: "normal",
      fontStyle: "normal",
      lineHeight: 1.45,
      letterSpacing: "normal",
      color: "#2a2a2a",
      marginBottom: "1rem",
    },
    subTitle: {
      fontWeight: "normal",
      fontStretch: "normal",
      fontStyle: "normal",
      lineHeight: 1,
      letterSpacing: "normal",
      color: "#474747",
      marginRight: "1rem",
    },
    formControl: {
      minWidth: 120,
      maxWidth: 240,
    },
    selectedItem: {
      color: theme.palette.primary.main,
    },
    menuItem: {
      color: theme.palette.primary.main,
    },
    textField: {
      width: "100%",
      overflow: "hidden",
      textOverflow: "ellipsis",
      display: "block",
    },
    pointDeleteButton: {
      marginLeft: "1rem",
    },
  });

interface SurveyIncomingProps {
  reefId: number;
}

type SurveysProps = SurveyIncomingProps & WithStyles<typeof styles>;

export default withStyles(styles)(Surveys);
