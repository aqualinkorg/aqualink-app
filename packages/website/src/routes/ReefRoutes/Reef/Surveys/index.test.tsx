import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";

import Surveys from ".";
import { mockUser } from "../../../../mocks/mockUser";
import { mockSurvey } from "../../../../mocks/mockSurvey";

const mockStore = configureStore([]);

describe("ReefRoutes Surveys", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      surveyList: {
        list: [mockSurvey],
        loading: false,
        error: null,
      },
      user: {
        userInfo: mockUser,
        loading: false,
        error: null,
      },
    });

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <Router>
          <Surveys reefId={0} />
        </Router>
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
