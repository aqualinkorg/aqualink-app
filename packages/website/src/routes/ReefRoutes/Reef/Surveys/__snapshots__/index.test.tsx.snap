// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReefRoutes Surveys should render with given state from Redux store 1`] = `
<div>
  <mock-dialog
    open="false"
  >
    <mock-dialogtitle
      classname="Dialog-dialogTitle-10"
    >
      <mock-typography>
        Are you sure you want to delete this survey point? It will be deleted across all surveys.
      </mock-typography>
    </mock-dialogtitle>
    <mock-dialogcontent />
    <mock-dialogactions>
      <mock-button
        color="secondary"
        size="small"
        variant="contained"
      >
        No
      </mock-button>
      <mock-button
        color="primary"
        size="small"
        variant="contained"
      >
        Yes
      </mock-button>
    </mock-dialogactions>
  </mock-dialog>
  <div
    class="MuiGrid-root Surveys-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
  >
    <mock-box
      bgcolor="#f5f6f6"
      height="100%"
      position="absolute"
      width="99vw"
      zindex="-1"
    />
    <div
      class="MuiGrid-root Surveys-surveyWrapper-2 MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-baseline MuiGrid-justify-xs-space-between MuiGrid-grid-xs-11 MuiGrid-grid-lg-12"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-md-12 MuiGrid-grid-lg-4"
      >
        <mock-typography
          classname="Surveys-title-3"
        >
          Survey History
        </mock-typography>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-md-12 MuiGrid-grid-lg-4"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <mock-typography
            classname="Surveys-subTitle-4"
            variant="h6"
          >
            Survey Point:
          </mock-typography>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-md-12 MuiGrid-grid-lg-4"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <mock-typography
            classname="Surveys-subTitle-4"
            variant="h6"
          >
            Observation:
          </mock-typography>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiFormControl-root Surveys-formControl-5"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline Surveys-selectedItem-6 MuiInputBase-formControl MuiInput-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="survey-observation survey-observation"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Surveys-textField-8"
                id="survey-observation"
                role="button"
                tabindex="0"
              >
                <mock-typography
                  classname="Surveys-menuItem-7"
                  variant="h6"
                >
                  Any
                </mock-typography>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                name="survey-observation"
                tabindex="-1"
                value="any"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <mock-menu
                anchorel="[object HTMLDivElement]"
                id="menu-survey-observation"
                menulistprops="[object Object]"
                open="false"
                paperprops="[object Object]"
              >
                <mock-menuitem
                  aria-selected="true"
                  data-value="any"
                  role="option"
                  selected="true"
                >
                  <mock-typography
                    classname="Surveys-menuItem-7"
                    variant="h6"
                  >
                    Any
                  </mock-typography>
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-7"
                  data-value="healthy"
                  role="option"
                  selected="false"
                >
                  Appears healthy
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-7"
                  data-value="possible-disease"
                  role="option"
                  selected="false"
                >
                  Showing possible signs of disease
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-7"
                  data-value="evident-disease"
                  role="option"
                  selected="false"
                >
                  Signs of disease are evident
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-7"
                  data-value="mortality"
                  role="option"
                  selected="false"
                >
                  Signs of mortality are evident
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-7"
                  data-value="environmental"
                  role="option"
                  selected="false"
                >
                  Signs of environmental disturbance are evident (e.g.,storm damage)
                </mock-menuitem>
                <mock-menuitem
                  classname="Surveys-menuItem-7"
                  data-value="anthropogenic"
                  role="option"
                  selected="false"
                >
                  Signs of anthropogenic disturbance are evident (e.g.,fishing gear, trampling)
                </mock-menuitem>
              </mock-menu>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-11 MuiGrid-grid-lg-12"
    >
      <div
        class="SurveyTimeline-root-20"
      >
        <mock-hidden
          mddown="true"
        >
          <ul
            class="MuiTimeline-root MuiTimeline-alignLeft"
          />
        </mock-hidden>
        <mock-hidden
          lgup="true"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
          />
        </mock-hidden>
      </div>
    </div>
  </div>
</div>
`;
